packages/calling/calling/src/CallingClient/registration/register.test.ts

```diff
@@ -1,111 +1,1013 @@
+
+      /* Mocking clearInterval because fakeTimers can't clear real timers */
+      jest.spyOn(global, 'clearInterval').mockReturnValue();
+
+      /* Set mercury connection to be down and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = false;
+      jest.advanceTimersByTime(2500);
+
+      /* We should be detecting the network flap */
+      expect(warnSpy).toHaveBeenCalledWith(
+        'Network has flapped, waiting for mercury connection to be up',
+        {file: REGISTRATION_FILE, method: NETWORK_CHANGE_DETECTION_UTIL}
+      );
+
+      /* Set mercury connection to be up and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = true;
+      jest.advanceTimersByTime(2500);
+
+      await flushPromises();
+
+      /* We should be detecting the network recovery */
+      expect(logSpy).toHaveBeenCalledWith(
+        'Mercury connection is up again, Re-registering with Mobius',
+        {file: REGISTRATION_FILE, method: 'handleConnectionRestoration'}
+      );
+
+      /*
+       * When initial registration is not done, network flap
+       * will not trigger de-registration/registration
+       */
+      expect(webex.request).not.toBeCalled();
+      expect(restoreSpy).not.toBeCalled();
+      expect(registerSpy).not.toBeCalled();
     });
 
-    global.fetch = jest.fn(() => Promise.resolve({json: () => mockResponseBody})) as jest.Mock;
+    it('Simulate a network flap with 1 active call', async () => {
+      const registerSpy = jest.spyOn(reg as any, 'attemptRegistrationWithServers');
+
+      /** create a new call */
+      reg.callManager.createCall();
+      expect(Object.keys(reg.callManager.getActiveCalls()).length).toBe(1);
 
-    const response = await reg.deleteDevice(URL, DEVICE_ID, CLIENT_DEVICE_URL);
+      /* Set mercury connection to be down and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = false;
+      jest.advanceTimersByTime(2500);
+      await flushPromises();
 
-    expect(response).toStrictEqual(mockResponseBody);
+      /* We should be detecting the network flap */
+      expect(warnSpy).not.toHaveBeenCalledWith(
+        'Network has flapped, waiting for mercury connection to be up',
+        {file: REGISTRATION_FILE, method: 'handleConnectionRestoration'}
+      );
+
+      /* Set mercury connection to be up and execute a delay of 2.5 seconds */
+      webex.internal.mercury.connected = true;
+      jest.advanceTimersByTime(2500);
+
+      await flushPromises();
+
+      /* We should be detecting the network recovery */
+      expect(logSpy).not.toHaveBeenCalledWith(
+        'Mercury connection is up again, Re-registering with Mobius',
+        {file: REGISTRATION_FILE, method: 'handleConnectionRestoration'}
+      );
+
+      expect(registerSpy).not.toHaveBeenCalledWith(true);
+      expect(registerSpy).toHaveBeenCalledTimes(0);
+    });
   });
 });
+
+/**
+ * 
+it('create device', async () => {
```

Comment:
Why are these UTs commented?

Reply:
They were old ones. No need to have them now with new tests